File: backend/src/main/java/coursepick/coursepick/domain/course/CourseName.java
package coursepick.coursepick.domain.course;

import static coursepick.coursepick.application.exception.ErrorType.INVALID_NAME_LENGTH;

public record CourseName(
        String value
) {
    public CourseName {
        value = compact(value);
        validateLength(value);
    }

    private static String compact(String value) {
        return value.trim().replaceAll("\\s+", " ");
    }

    private static void validateLength(String compactName) {
        if (compactName.length() < 2 || compactName.length() > 30) {
            throw INVALID_NAME_LENGTH.create(compactName);
        }
    }
}

File: android/app/src/main/java/io/coursepick/coursepick/domain/course/CourseName.kt
package io.coursepick.coursepick.domain.course

@JvmInline
value class CourseName(
    val value: String,
) {
    init {
        require(value.length in MIN_LENGTH..MAX_LENGTH)
    }

    companion object {
        const val MIN_LENGTH = 2
        const val MAX_LENGTH = 30
    }
}
